### ISSUE TITLE
possible to generate a header navigation bar to the rmarkdown website?

### ISSUE THREAD

**Issue by sculptor3** [yutannihilation/rmarkdown-website-template#1]
Hi, 
Many thanks for this light version of the RStudio rmarkdown website generation. It has been proven really helpful. I know it's a light version, but I was wondering, is it possible to add a funcionality of generating a header navigation bar (like the one of RStudio)? In the RStudio website it's stored in the include directory, named 'before_html'?

kind regards

**Comment by yutannihilation** [comment]
Thanks for your comment!

Now I added a simple navigation bar. Is this what you want? Or, do you want to generate the navigation bar which includes all the pages automatically?

https://github.com/yutannihilation/rmarkdown-website-template/pull/2

![header](https://cloud.githubusercontent.com/assets/1978793/9964618/545f7d5e-5e6c-11e5-8e2e-54427c14fada.PNG)

**Comment by sculptor3** [comment]
Thanks for your quick esponse! This looks promising. I guess I meant to
generate it automatically in much the same way as the index.html page is
generated (with your r script), but I think this will do as well! I will
take a look.
Op 18 sep. 2015 18:19 schreef "yutannihilation" notifications@github.com:

**Comment by sculptor3** [comment]
HI,

I did some experiments. The navigation bar you added is very helpful.
Thanks again!
I think your template would be more powerful when the generateIndex.R
script would not only generate the index.html page but also the
before_body.html file including the toc's of the pages as dropdowns. In
fact that would be even better then just adding the html-pages as links in
the index.html body. To my opinion the content of the index.html page
itself should be arbitrary, and filled by whatever the site builder wants
to put there (but maybe that's just a matter of taste :-)).

Thanks again for your very helpful template!

Kind regards,
Jelle

2015-09-18 20:19 GMT+02:00 Jelle Geertsma jelle.geertsma@gmail.com:

**Comment by yutannihilation** [comment]
My personal goal was to create some kind of [RPubs](https://rpubs.com/) clone, which may be different from what you expect for this repo... (sorry for not writing this in README)

But I like your idea! :+1: I will try!

> To my opinion the content of the index.html page itself should be arbitrary, and filled by whatever the site builder wants to put there (but maybe that's just a matter of taste :-)).

**Comment by yutannihilation** [comment]
I created a simple dropdown list of pages. How about this?

![header2](https://cloud.githubusercontent.com/assets/1978793/9973954/dd627dee-5eb8-11e5-8e5b-6f8f889ce7ab.PNG)

Now this repository uses these three scripts:
- `index.R`: generate `index.json`, which contains metadata of pages. Other script uses `index.json` to generate the page list.
- `generateNavigationBar.R`: generate `include/before_body.html`, which is included in each knitted pages as a navigation bar.
- `generateIndexPage`: generate `index.html`.

If you want to use your own `index.html`, you can modify Makefile not to generate index.html automatically like this:

https://github.com/yutannihilation/rmarkdown-website-template/blob/gh-pages/Makefile#L9

``` diff
- all: index generateNavigationBar html generateIndexPage
+ all: index generateNavigationBar html
```

**Comment by sculptor3** [comment]
this is brilliant :-). And I again I thank you very much. After some trials I think I got it working. I guess this should be the order of generating: 1. index.R, 2. generateNavitaionBar.R, 3 generateIndexPage?

My first thought was to give each page it's own dropdown with the table of content of that page as dropdown content (if there is a table of content ofcourse). But I guess I can figure that out myself somehow. I will keep on experimenting.

Thanks again!

**Comment by yutannihilation** [comment]
Thanks for checking!

> I guess this should be the order of generating: 1. index.R, 2. generateNavitaionBar.R, 3 generateIndexPage?

Yes. More precisely,
1. execute `index.R` to generate `index.json`
2. execute `generateNavigationBar.R` to generate `before_body.html`
3. execute `rmarkdown::render()` to knit all .Rmd files
4. execute `generateIndexPage.html` to generate `index.html`.

> My first thought was to give each page it's own dropdown with the table of content of that page as dropdown content (if there is a table of content ofcourse).

For getting metadata in YAML front-matter, luckily, I found `rmarkdown:::parse_yaml_front_matter()` (which is not exported). But, sorry, I could not find any convenient function to generate ToC... :cry: 

Anyway, I wish you good luck!

**Comment by sculptor3** [comment]
It works perfectly for me now. I altered the navigation bar generator
slightly. Instead of letting it put the links to the pages in a dropdown, I
let it put them next to eachother on the bar.

To my opinion is this a poweful template and very useful for generating a
complete website based on rmarkdown files!  So the issue can be closed I
guess.

My next goal is to make the template more look like a dashboard by
implementing twitter bootstrap grid layout and other dashboard features.

Thanks for checking!

I guess this should be the order of generating: 1. index.R, 2.
generateNavitaionBar.R, 3 generateIndexPage?

Yes. More precisely,
1. execute index.R to generate index.json
2. execute generateNavigationBar.R to generate before_body.html
3. execute rmarkdown::render() to knit all .Rmd files
4. execute generateIndexPage.html to generate index.html.

My first thought was to give each page it's own dropdown with the table of
content of that page as dropdown content (if there is a table of content
ofcourse).

For getting metadata in YAML front-matter, luckily, I found
rmarkdown:::parse_yaml_front_matter() (which is not exported). But, sorry,
I could not find any convenient function to generate ToC... [image: :cry:]

Anyway, I wish you good luck!
